fedimint-server/src/net/peers.rs

```diff
@@ -692,36 +479,45 @@ where
         our_id: PeerId,
         peer_id: PeerId,
         peer_address: SafeUrl,
-        delay_calculator: DelayCalculator,
         connect: SharedAnyConnector<PeerMessage<M>>,
         incoming_connections: Receiver<AnyFramedTransport<PeerMessage<M>>>,
         status_channels: Arc<RwLock<BTreeMap<PeerId, PeerConnectionStatus>>>,
         task_handle: &TaskHandle,
     ) {
-        let common = CommonPeerConnectionState {
-            incoming,
-            outgoing,
-            our_id_str: our_id.to_string(),
-            our_id,
-            peer_id_str: peer_id.to_string(),
-            peer_id,
-            peer_address,
-            delay_calculator,
-            connect,
-            incoming_connections,
-            status_channels,
+        info!(target: LOG_NET_PEER, "Starting peer connection state machine {}", peer_id);
+
+        let mut state_machine = PeerConnectionStateMachine {
+            common: CommonPeerConnectionState {
+                incoming,
+                outgoing,
+                our_id_str: our_id.to_string(),
+                our_id,
+                peer_id_str: peer_id.to_string(),
+                peer_id,
+                peer_address,
+                connect,
+                incoming_connections,
+                status_channels,
+            },
+            state: PeerConnectionState::Disconnected(api_networking_backoff()),
         };
-        let initial_state = PeerConnectionState::Disconnected(DisconnectedPeerConnectionState {
-            reconnect_at: Instant::now(),
-            failed_reconnect_counter: 0,
-        });
 
-        let state_machine = PeerConnectionStateMachine {
-            common,
-            state: initial_state,
-        };
+        loop {
+            tokio::select! {
+                sm = state_machine.state_transition() => {
+                    if let Some(sm) = sm {
+                        state_machine = sm;
+                    } else {
+                        break;
+                    }
+                }
+                () = task_handle.make_shutdown_rx() => {
+                    break;
+                },
```

Comment:
can be removed imo if the io thread was spawned as cancellable

Reply:
Done